pkg/distro/generic/options_test.go

```diff
@@ -0,0 +1,1671 @@
+			it: "workstation-live-installer",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					User: []blueprint.UserCustomization{{Name: "root"}},
+					Installer: &blueprint.InstallerCustomization{
+						Unattended: true,
+					},
+				},
+			},
+			expErr: "image type \"workstation-live-installer\" does not support customizations",
+		},
+
+		"ostree-types-no-oscap": {
+			it: "iot-container",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					OpenSCAP: &blueprint.OpenSCAPCustomization{
+						ProfileID: "xccdf_org.ssgproject.content_profile_ospp",
+					},
+				},
+			},
+			expErr: "OpenSCAP customizations are not supported for ostree types",
+		},
+
+		"iot-installer-installer-customizations": {
+			it: "iot-installer",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					Installer: &blueprint.InstallerCustomization{
+						Unattended: true,
+					},
+				},
+			},
+			options: distro.ImageOptions{
+				OSTree: &ostree.ImageOptions{
+					URL: "https://example.org/repo",
+				},
+			},
+		},
+		"iot-installer-bad-combinations": {
+			it: "iot-installer",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					User: []blueprint.UserCustomization{{Name: "root"}},
+					Installer: &blueprint.InstallerCustomization{
+						Kickstart: &blueprint.Kickstart{
+							Contents: "echo 'Testing'",
+						},
+					},
+				},
+			},
+			options: distro.ImageOptions{
+				OSTree: &ostree.ImageOptions{
+					URL: "https://example.org/repo",
+				},
+			},
+			expErr: "iot-installer installer.kickstart.contents are not supported in combination with users or groups",
+		},
+
+		"ostree-disk-unsupported-containers": {
+			it: "iot-qcow2",
+			bp: blueprint.Blueprint{
+				Containers: []blueprint.Container{
+					{
+						Source: "example.org/containers/test:42",
+					},
+				},
+			},
+			expErr: "embedding containers is not supported for iot-qcow2 on fedora-42",
+		},
+
+		"ostree-commit-unsupported-kernel-append": {
+			it: "iot-commit",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					Kernel: &blueprint.KernelCustomization{
+						Append: "debug",
+					},
+				},
+			},
+			expErr: "kernel boot parameter customizations are not supported for ostree types",
+		},
+
+		"oscap-empty-profile": {
+			it: "vhd",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					OpenSCAP: &blueprint.OpenSCAPCustomization{
+						ProfileID: "",
+					},
+				},
+			},
+			expErr: "OpenSCAP profile cannot be empty",
+		},
+
+		// NOTE: the following tests verify the current behaviour of the
+		// function, but the behaviour itself is wrong
+		"ostree-disk-requires-ostree-url": {
+			it:     "iot-qcow2",
+			expErr: "", // NOTE: it should require a URL
+		},
+		"ostree-disk2-requires-ostree-url": {
+			it:     "iot-raw-xz",
+			expErr: "", // NOTE: it should require a URL
+		},
```

Comment:
I am not sure I understand, is this a leftover perhaps?

Reply:
Ooooh, ok.  Sorry, I misunderstood something in your earlier comment.  I think I completely missed the part about `checkOptionsCommon()`.  I'll fix this all up tomorrow.  Thanks!